### PR TITLE
[Documents] Newsletter: add option to maintain a manual edited plain-text version

### PR DESCRIPTION
**Pull request by Lexipowder**
Spam filters like SpamAssassin rate multipart emails higher than just html-email. Having the text part is worth one score point. The existing version just sends html-emails, also with the detour via html2text. (See comment in code).

Following database change has to be done first: (I don't know how to send the db-update in a PR)
ALTER TABLE `intranet`.`documents_newsletter` 
ADD COLUMN `plaintext` LONGTEXT NULL DEFAULT NULL AFTER `sendingMode`;

Best regards
Lexi

### CODE REVIEW

File: lib/Tool/Newsletter.php [hunk 1]
```diff
@@ -115,6 +119,8 @@ public static function prepareMail(
 
         $mail->setBodyHtml($contentHTML);
         $mail->setBodyText($contentText);
+        // Adds the plain text part to the message, that it becomes a multipart email
+        $mail->addPart($contentText, 'text/plain');
```
Comment (brusch) [inline lib/Tool/Newsletter.php]: Why? See: 
https://github.com/pimcore/pimcore/blob/d688a67780b6ae733e145110bbbc04b3278e7190/lib/Mail.php#L567

File: bundles/CoreBundle/Resources/translations/de.json [hunk 2]
```diff
@@ -52,6 +52,8 @@
     "send_newsletter": "Newsletter JETZT versenden",
     "object_filter": "Objekt Filter",
     "add_newsletter": "Newsletter hinzufügen",
+    "plain_text": "Klartext",
```
Comment (brusch) [inline bundles/CoreBundle/Resources/translations/de.json]: Languages other than EN are managed here: 
Please revert this change and add the translation in POEditor once this PR is merged.

File: lib/Tool/Newsletter.php [hunk 3]
```diff
@@ -137,8 +143,6 @@ public static function sendNewsletterDocumentBasedMail(Mail $mail, SendingParamC
 
         if (!empty($mailAddress)) {
             $mail->setTo($mailAddress);
-            // Getting bounces
```
[the remark this answers is not in the document]
Comment (Lexipowder) [reply lib/Tool/Newsletter.php]: Not necessary, as the SMTP Server will set it automatically set the return path when finally sending the email.

https://www.postmastery.com/about-the-return-path-header/

File: bundles/CoreBundle/Resources/translations/de.json (hunk 2 above)
Comment (Lexipowder) [reply bundles/CoreBundle/Resources/translations/de.json]: Where exactly? Link missing...

File: lib/Tool/Newsletter.php (hunk 1 above)
Comment (Lexipowder) [reply lib/Tool/Newsletter.php]: Line 567 is part of the send function, the newsletter is send via sendWithoutRendering function:

Pimcore::getEventDispatcher()->dispatch(DocumentEvents::NEWSLETTER_PRE_SEND, $event);
            $mail->sendWithoutRendering($mailer);
            Pimcore::getEventDispatcher()->dispatch(DocumentEvents::NEWSLETTER_POST_SEND, $event);

File: bundles/CoreBundle/Resources/translations/de.json (hunk 2 above)
Comment (brusch) [reply bundles/CoreBundle/Resources/translations/de.json]: https://poeditor.com/join/project/VWmZyvFVMH

image
image

File: lib/Tool/Newsletter.php (hunk 1 above)
Comment (brusch) [reply lib/Tool/Newsletter.php]: Good point 👍😊

File: lib/Tool/Newsletter.php (hunk 3 above)
[the remark this answers is not in the document]
Comment (brusch) [reply lib/Tool/Newsletter.php]: You added this in #4349 ... can't hurt to leave it as it is, right?

[the remark this answers is not in the document]
Comment (Lexipowder) [reply lib/Tool/Newsletter.php]: I know it was me. :-) 
But on what I read in the meantime, I would recommend to leave it out, because some mailserver seems to reject or identify the email as spam, when return path is wrongly set. At the end if we leave it in, it won't hurt the sending process, !maybe! some emails won't receive the customer.

File: bundles/CoreBundle/Resources/translations/de.json (hunk 2 above)
Comment (Lexipowder) [reply bundles/CoreBundle/Resources/translations/de.json]: Thank you, will do
